### PR TITLE
fix(nns): validate dissolve state and age before saving neuron

### PR DESCRIPTION
**Pull request by max-dfinity**
This fixes a minor discrepancy that would prevent certain malformed neurons from being retrieved from stable storage after they've been written, which would make those neurons inaccessible to users.  This would only affect neurons after active neurons are also stored in stable storage.

When a neuron's DissolveState was set to WhenDissolvedTimestampSeconds(0), the translation back and forth would result in a validation error, and thus a panic.  

This ensures that the field values that are written can also be read back before they are written.

### CODE REVIEW

File: rs/nns/test_utils/src/gtc_helpers.rs [hunk 1]
```diff
@@ -242,7 +242,7 @@ fn make_neuron(
 
     let (dissolve_state, aging_since_timestamp_seconds) = if dissolve_delay_seconds == 0 {
         (
-            Some(DissolveState::WhenDissolvedTimestampSeconds(0)),
+            Some(DissolveState::WhenDissolvedTimestampSeconds(1)),
```
Comment (max-dfinity) [inline rs/nns/test_utils/src/gtc_helpers.rs:245]: This means it's dissolved 1 second after linux epoch.  Should make no functional difference.

File: rs/nns/governance/src/neuron/dissolve_state_and_age.rs [hunk 2]
```diff
@@ -20,6 +22,22 @@ pub enum DissolveStateAndAge {
 }
 
 impl DissolveStateAndAge {
+    pub fn validate(self) -> Result<Self, String> {
```
Comment (max-dfinity) [inline rs/nns/governance/src/neuron/dissolve_state_and_age.rs:25]: Due to an edge case where neurons created could not be read from stable memory, we now attempt the conversion here, to ensure that the requirements are met before saving.

File: rs/nns/governance/src/neuron/dissolve_state_and_age.rs [hunk 3]
```diff
@@ -20,6 +22,22 @@ pub enum DissolveStateAndAge {
...
+        let original = self;
+        let stored_dissolve_state_and_age = StoredDissolveStateAndAge::from(self);
+
+        let validated_dissolve_state_and_age = Self::try_from(stored_dissolve_state_and_age)
+            .map_err(|e| format!("Invalid dissolve state and age: {}", e))?;
+
+        if validated_dissolve_state_and_age != original {
+            return Err( format!(
+                    "Dissolve state and age is not valid, as it does not serialize/deserialize symmetrically. In: {:?}, Out: {:?}",
```
Comment (daniel-wong-dfinity-org) [inline rs/nns/governance/src/neuron/dissolve_state_and_age.rs:34]: Not sure this is uses standard meanings of "symmetric". I would say something more like, "Conversion round trip did not result in an equal value.".

File: rs/nns/governance/src/neuron/dissolve_state_and_age.rs (hunk 2 above)
Comment (daniel-wong-dfinity-org) [reply rs/nns/governance/src/neuron/dissolve_state_and_age.rs:25]: Another example of vague: "an edge case". Questions that should probably be addressed: What are the conditions that lead to bad things? What bad things result from those conditions such that neurons cannot be read from stable memory?

**Review by daniel-wong-dfinity-org — DISMISSED** [review]
Title and description seem vague. E.g. "certain malformed neurons". Which ones? <- seems like a pretty salient question. Description should answer salient questions.

TBH, I'm really struggling to read this without such details.

**Review by anchpop — APPROVED** [review]

**Pull request merged**